src/tdastro/astro_utils/dust_map.py

```diff
@@ -0,0 +1,168 @@
+from astropy.coordinates import SkyCoord
+from dustmaps.config import config as dm_config
+
+from tdastro import _TDASTRO_CACHE_DATA_DIR
+
+
+class DustExtinctionEffect:
+    """A general dust extinction model.
+
+    Attributes
+    ----------
+    dust_map : dustmaps.DustMap or str
+        The dust map or its name. Since different dustmap's query function
+        may produce different outputs, you should include the corresponding
+        ebv_func to transform the result into ebv if needed.
+    extinction_model : function or str
+        The extinction object from the dust_extinction library or its name.
+        If a string is provided, the code will find a matching extinction
+        function in the dust_extinction package and use that.
+    ebv_func : function
+        A function to translate the result of the dustmap query into an ebv.
+    **kwargs : `dict`, optional
+        Any additional keyword arguments.
+    """
+
+    def __init__(self, dust_map, extinction_model, ebv_func=None, **kwargs):
+        self.ebv_func = ebv_func
+
+        if isinstance(dust_map, str):
+            # Initially we only support loading the SFD dustmap by string.
+            # But we can expand this as needed.
+            dust_map = dust_map.lower()
+            if dust_map == "sfd":
+                self._load_sfd_dustmap()
+            else:
+                raise ValueError("Unsupported load from dustmap {dust_map}")
+        else:
+            # If given a dustmap, use that directly.
+            self.dust_map = dust_map
+
+        if isinstance(extinction_model, str):
+            extinction_model = DustExtinctionEffect.load_extinction_model(extinction_model, **kwargs)
+        self.extinction_model = extinction_model
+
+    def _load_sfd_dustmap(self):
+        """Load the SFD dustmap, downloading it if needed.
+
+        Uses data from:
+        1.  Schlegel, Finkbeiner, and Davis
+            The Astrophysical Journal, Volume 500, Issue 2, pp. 525-553.
+            https://ui.adsabs.harvard.edu/abs/1998ApJ...500..525S/abstract
+
+        2.  Schlegel and Finkbeiner
+            The Astrophysical Journal, Volume 737, Issue 2, article id. 103, 13 pp. (2011).
+            https://ui.adsabs.harvard.edu/abs/2011ApJ...737..103S/abstract
+
+        Returns
+        -------
+        sfd_query : SFDQuery
+            The "query" object for the requested dustmap.
+        """
+        import dustmaps.sfd
+
+        # Download the dustmap if needed.
+        dm_config["data_dir"] = str(_TDASTRO_CACHE_DATA_DIR / "dustmaps")
+        dustmaps.sfd.fetch()
+
+        # Load the dustmap.
+        from dustmaps.sfd import SFDQuery
+
+        self.dust_map = SFDQuery()
+
+        # Add the correction function.
+        def _sfd_scale_ebv(input, **kwargs):
+            """Scale the result of the SFD query."""
+
+        self.ebv_func = _sfd_scale_ebv
+
+    @staticmethod
+    def load_extinction_model(name, **kwargs):
+        """Load the extinction model.
```

Comment:
I suggest add source of the extinction model for users who are not familiar with the package.
"Load the extinction model from the dust_extinction package (https://github.com/karllark/dust_extinction)"